selfservice/strategy/oidc/strategy.go

```diff
@@ -291,29 +246,47 @@ func (s *Strategy) validateFlow(ctx context.Context, r *http.Request, rid uuid.U
 	return ar, err // this must return the error
 }
 
-func (s *Strategy) ValidateCallback(w http.ResponseWriter, r *http.Request) (flow.Flow, *AuthCodeContainer, error) {
+func (s *Strategy) ValidateCallback(w http.ResponseWriter, r *http.Request) (f contextFlow, providerID string, ac *AuthCodeContainer, err error) {
 	var (
 		codeParam  = stringsx.Coalesce(r.URL.Query().Get("code"), r.URL.Query().Get("authCode"))
 		stateParam = r.URL.Query().Get("state")
 		errorParam = r.URL.Query().Get("error")
 	)
 
 	if stateParam == "" {
-		return nil, nil, errors.WithStack(herodot.ErrBadRequest.WithReasonf(`Unable to complete OpenID Connect flow because the OpenID Provider did not return the state query parameter.`))
+		return nil, "", nil, errors.WithStack(herodot.ErrBadRequest.WithReasonf(`Unable to complete OpenID Connect flow because the OpenID Provider did not return the state query parameter.`))
 	}
 	state, err := parseState(stateParam)
 	if err != nil {
-		return nil, nil, errors.WithStack(herodot.ErrBadRequest.WithReasonf(`Unable to complete OpenID Connect flow because the state parameter was invalid.`))
+		return nil, "", nil, errors.WithStack(herodot.ErrBadRequest.WithReasonf(`Unable to complete OpenID Connect flow because the state parameter was invalid.`))
 	}
 
-	f, err := s.validateFlow(r.Context(), r, x.ParseUUID(state.FlowID))
+	f, err = s.validateFlow(r.Context(), r, x.ParseUUID(state.FlowID))
 	if err != nil {
-		return nil, nil, err
+		return nil, "", nil, err
+	}
+
+	// Determine the provider from the flow context or the URL.
+	providerID = providerFromFlow(f)
+	urlParams, _ := r.Context().Value(httprouter.ParamsKey).(httprouter.Params)
```

Comment:
Please pass this as an argument instead

Reply:
Is it not possible to pass this value as an argument? The method is only used once afaik:

```
ValidateCallback(w http.ResponseWriter, r *http.Request, params httprouter.Params)
```

Reply:
Yes, it's possible. Will address.